project/src/analysis/sampler.py

```diff
@@ -0,0 +1,36 @@
+"""Use the pymbar package to perform decorrelated equilibration sampling."""
+
+import numpy as np
+from pymbar import timeseries
+
+from project.src.analysis.equlibration import is_equilibrated
+
+
+def decorr_sampling(job, data_type="potential_energy"):
+    """Use the timeseries module from pymbar to perform statistical sampling.
+
+    The start, end and decorrleated step size of the production region are
+    added to the job document.
+
+    Parameters
+    ----------
+    job : signac.contrib.job.Job
+        The Job object.
+    data_type : str; default "potential_energy"
+        The type of simulation data to be used in sampling
+    """
+    data = np.genfromtxt(job.fn("log.txt", names=True))[data_type]
+    is_equil, prod_start, ineff = is_equilibrated(data, threshold=0.8, nskip=1)
+    if is_equil:
+        uncorr_indices = timeseries.subsampleCorrelatedData(
+            data[prod_start:], g=ineff, conservative=True
+        )
+        job.doc["sample_start"] = uncorr_indices.start + prod_start
+        job.doc["sample_end"] = uncorr_indices.stop + prod_start
+        job.doc["sample_stride"] = uncorr_indices.step
```

Comment:
Based on https://pymbar.readthedocs.io/en/master/timeseries.html#pymbar.timeseries.subsampleCorrelatedData

I am not sure where the uncorr_indices.start, stop, step come from? `subsampleCorrelatedData` does not seem to return that kind of information?

Reply:
`subsampleCorrelatedData` returns a range--not a list as said in the docs. Anyway, this allows us to access the start, stop, and step info.

Reply:
It depends if `conservative` is set to `True` or `False`. When `True` it returns a `range`, when it's `False` it will return a list

Reply:
Ah OK, I didn't realize that.

Reply:
Great catch, I missed that and I assumed that `range` would return a list, but that does not make sense since it is a range object.